op/model.py

```diff
@@ -229,6 +231,28 @@ def _load(self):
         return self._backend.config_get()
 
 
+class ResourceMapping(Mapping):
+    """Map of resource names to the local path for the resource."""
+    def __init__(self, resource_names, backend):
+        self._backend = backend
+        self._data = {resource_name: None for resource_name in resource_names}
+
+    def __contains__(self, key):
+        return key in self._data
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __getitem__(self, resource_name):
+        resource_path = self._data[resource_name]
+        if resource_path is None:
+            resource_path = self._data[resource_name] = Path(self._backend.resource_get(resource_name))
```

Comment:
Doesn't that involve downloading a file, potentially a large one?  A common anti-pattern is pretending an expensive operation is cheap. For example, nobody will expect doing self.resources["foo"] might take 5 minutes to return.

Reply:
Yeah, I was really on the fence as to whether it was better to have the fetch be more explicit rather than just matching the style we used for the other objects, so I'll go ahead and change it.  That will sort out the other comment as well.